Source/Core/Core/PowerPC/JitCommon/DivUtils.cpp

```diff
@@ -44,13 +44,47 @@ Magic SignedDivisionConstants(s32 d)
     delta = ad - r2;
   } while (q1 < delta || (q1 == delta && r1 == 0));
 
-  Magic mag;
+  SignedMagic mag;
   mag.multiplier = q2 + 1;
-  if (d < 0)
+  if (divisor < 0)
     mag.multiplier = -mag.multiplier;
   mag.shift = p - 32;
 
   return mag;
 }
 
+UnsignedMagic UnsignedDivisionConstants(u32 divisor)
+{
+  u32 shift = 31;
+  while (!(divisor & (1 << shift)))
+    shift--;
+
+  u64 magic_dividend = 0x100000000ULL << shift;
+  u32 multiplier = (u32)(magic_dividend / divisor);
+  u32 max_quotient = multiplier >> shift;
+
+  // Test for failure in round-up method
+  u32 round_up = ((u64)(multiplier + 1) * (max_quotient * divisor - 1)) >> (shift + 32);
+  bool fast = round_up == max_quotient - 1;
+
+  if (fast)
+  {
+    multiplier++;
+
+    // Use smallest magic number and shift amount possible
+    while ((multiplier & 1) == 0 && shift > 0)
+    {
+      multiplier >>= 1;
+      shift--;
+    }
```

Comment:
```suggestion
    const int x = std::min(shift, std::countr_zero(multiplier));
    multiplier >>= x;
    shift -= x;
```

(`x` is a bad name, please do replace it if you can think of something better)